OpenRA.Game/Graphics/UISpriteRenderable.cs

```diff
@@ -22,8 +22,9 @@ public class UISpriteRenderable : IRenderable, IPalettedRenderable, IFinalizedRe
 		readonly PaletteReference palette;
 		readonly float scale;
 		readonly float alpha;
+		readonly float rotation = 0f;
 
-		public UISpriteRenderable(Sprite sprite, WPos effectiveWorldPos, int2 screenPos, int zOffset, PaletteReference palette, float scale = 1f, float alpha = 1f)
+		public UISpriteRenderable(Sprite sprite, WPos effectiveWorldPos, int2 screenPos, int zOffset, PaletteReference palette, float scale, float alpha, float rotation)
```

Comment:
Because it is otherwise not possible to pass in rotation, since parameters with default values must always be last (scale and alpha above having default value of 1f).

Reply:
?

Reply:
https://www.geeksforgeeks.org/c-sharp-optional-parameters/

>The optional parameters are always defined at the end of the parameter list. Or in other words, the last parameter of the method, constructor, etc. is the optional parameter.

Reply:
Ok this should actually work with the default parameter `rotation = 0f` at the end. Have updated the code.